src/core/plugins/CMakeLists.txt

```diff
@@ -0,0 +1,57 @@
+#
```

Comment:
I’m not convinced by the use of “plugin” but I don’t actually haves much better proposition right now. Perhaps you could think about it once more?

Secondly, why not also move the built-in topic ones here?

Reply:
I was originally thinking about making custom WHC/RHC in cycloendds and then rmw_cyclonedds_cpp links to cyclonedds project. So create a *plugin* folder, it contains default WHC/RHC and simplified WHC/RHC methods (a simplified method without instance and key concept.) 
But I later realize this simplified WHC/RHC should be done in rmw_cyclonedds_cpp project to have a better integration with rmw layer. (for example rmw allocator..)

So I will move back the dds_whc_default.c and dds_rhc_default.c to *src/core/ddsc/src* in later change.


> Secondly, why not also move the built-in topic ones here?

I don't understand why built-in topic might be proper to put in plugin folder.